### PR TITLE
bpo-45548: Remove _math.c workarounds for pre-C99 libm

### PR DESCRIPTION
**Pull request by tiran**
The :mod:`math` and :mod:`cmath` implementation now require a C99 compatible
``libm`` and no longer ship with workarounds for missing acosh, asinh,
expm1, and log1p functions.

The changeset also removes ``_math.c`` and moves the last remaining
workaround into ``_math.h``. This simplifies static builds with
``Modules/Setup`` and resolves symbol conflicts.

Co-authored-by: Mark Dickinson <mdickinson@enthought.com>
Co-authored-by: Brett Cannon <brett@python.org>
Signed-off-by: Christian Heimes <christian@python.org>

https://bugs.python.org/issue45548

Automerge-Triggered-By: GH:tiran

### CODE REVIEW

**Comment by brettcannon** [comment]
I created an alternative approach but just inlining into `_math.h`: https://github.com/python/cpython/pull/29181 .

File: Misc/NEWS.d/next/Build/2021-10-24-21-49-49.bpo-45548.UWx0UC.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+The :mod:`math` and :mod:`cmath` implementation now require a C99 compatible
+``libm`` and no longer ship with workarounds for missing acosh, asinh,
```
Comment (mdickinson) [inline Misc/NEWS.d/next/Build/2021-10-24-21-49-49.bpo-45548.UWx0UC.rst]:
```suggestion
``libm`` and no longer ship with workarounds for missing acosh, asinh, atanh,
```

File: Modules/mathmodule.c [hunk 2]
```diff
@@ -61,6 +61,9 @@ raised for division by zero and mod by zero.
 #include "pycore_call.h"          // _PyObject_CallNoArgs()
 #include "pycore_dtoa.h"          // _Py_dg_infinity()
 #include "pycore_long.h"          // _PyLong_GetZero()
+/* For DBL_EPSILON in _math.c */
```
Comment (mdickinson) [inline Modules/mathmodule.c]: Comment needs an update?

File: setup.py [hunk 3]
```diff
@@ -904,18 +904,14 @@ def detect_simple_extensions(self):
         # Context Variables
         self.add(Extension('_contextvars', ['_contextvarsmodule.c']))
 
-        shared_math = 'Modules/_math.o'
-
         # math library functions, e.g. sin()
         self.add(Extension('math',  ['mathmodule.c'],
-                           extra_objects=[shared_math],
-                           depends=['_math.h', shared_math],
+                           depends=['_math.h', '_math.c'],
```
Comment (mdickinson) [inline setup.py]: Should `_math.c` be removed here (and below)?

**Review by mdickinson — APPROVED** [review]
Mostly LGTM; two nitpicks and one question.

File: Modules/mathmodule.c (hunk 2 above)
Comment (tiran) [reply Modules/mathmodule.c]: Thanks, I replace ``_math.c`` with ``_math.h``.

File: setup.py (hunk 3 above)
Comment (tiran) [reply setup.py]: You are correct!

**Pull request merged**